WordPress/gutenberg
# edit-navigation: migrate resolvers to thunks

This PR migrates the edit-navigation data store to thunks, namely the `getNavigationPostForMenu` resolver. It also  dependency on `@wordpress/data-controls` and on any other controls for that matter.

I reworked the unit tests to check for outcomes instead of the specific flow we tested for inside of the generator.

**How to test:**
Open the experimental navigation editor and confirm it loads your menus correctly.
